include/nuttx/atomic.h

```diff
@@ -81,16 +82,55 @@ extern "C++"
 #    if !(__clang__) && defined(__cplusplus)
 #      define _Atomic
 #    endif
-#    include <stdbool.h>
 #    include <stdatomic.h>
-#    ifndef ATOMIC_VAR_INIT
-#      define ATOMIC_VAR_INIT(value) (value)
-#    endif
-#  else
-#    include <nuttx/lib/stdatomic.h>
+#    define USE_STD_ATOMIC 1
 #  endif
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef __ATOMIC_RELAXED
+#  define __ATOMIC_RELAXED 0
+#endif
+
+#ifndef __ATOMIC_CONSUME
+#  define __ATOMIC_CONSUME 1
+#endif
+
+#ifndef __ATOMIC_ACQUIRE
+#  define __ATOMIC_ACQUIRE 2
+#endif
+
+#ifndef __ATOMIC_RELEASE
+#  define __ATOMIC_RELEASE 3
+#endif
+
+#ifndef __ATOMIC_ACQ_REL
+#  define __ATOMIC_ACQ_REL 4
+#endif
+
+#ifndef __ATOMIC_SEQ_CST
+#  define __ATOMIC_SEQ_CST 5
+#endif
+
+#ifdef USE_STD_ATOMIC
+#  define ATOMIC_FUNC(f, n) atomic_##f##_explicit
 #else
-#  include <nuttx/lib/stdatomic.h>
+#  define ATOMIC_FUNC(f, n) nx_atomic_##f##_##n
+#endif
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+#if __clang__
```

Comment:
when build c file, need add _Atomic 
error: address argument to atomic operation must be a pointer to _Atomic type ('atomic_t *' (aka 'volatile int *') invalid)

Reply:
but we can always typedef:
```
typedef volatile _Atomic int32_t atomic_t since line 83 already define an empty _Atomic:
typedef volatile _Atomic int64_t atomic64_t;
```